File: CS/SchedulerDbAccessExample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.XtraScheduler;

namespace SchedulerDbAccessExample {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            #region #SaveDataEvent
            // ...
            schedulerStorage1.AppointmentsChanged += OnAppointmentChangedInsertedDeleted;
            schedulerStorage1.AppointmentsInserted += OnAppointmentChangedInsertedDeleted;
            schedulerStorage1.AppointmentsDeleted += OnAppointmentChangedInsertedDeleted;
            // ...
            #endregion #SaveDataEvent
            #region #RowUpdatedEvent
            // ...
            appointmentsTableAdapter.Adapter.RowUpdated += new System.Data.OleDb.OleDbRowUpdatedEventHandler(appointmentsTableAdapter_RowUpdated);
            // ...
            #endregion #RowUpdatedEvent
            schedulerControl1.Start = DateTime.Today;
            schedulerControl1.GroupType = SchedulerGroupType.Resource;
            schedulerControl1.ActiveView.ResourcesPerPage = 2; ;
            schedulerStorage1.Appointments.ResourceSharing = true;

            schedulerControl1.OptionsView.ResourceHeaders.ImageAlign = DevExpress.XtraScheduler.HeaderImageAlign.Top;
            schedulerControl1.OptionsView.ResourceHeaders.ImageSizeMode = DevExpress.XtraScheduler.HeaderImageSizeMode.ZoomImage;
            schedulerControl1.OptionsView.ResourceHeaders.ImageSize = new Size(140, 70);
            schedulerControl1.OptionsView.ResourceHeaders.Height = 100;
        }


        #region #LoadData
        private void Form1_Load(object sender, EventArgs e) {
            // TODO: This line of code loads data into the 'schedulerTestDataSet.Resources' table. You can move, or remove it, as needed.
            this.resourcesTableAdapter.Fill(this.schedulerTestDataSet.Resources);
            // TODO: This line of code loads data into the 'schedulerTestDataSet.Appointments' table. You can move, or remove it, as needed.
            this.appointmentsTableAdapter.Fill(this.schedulerTestDataSet.Appointments);
        }
        #endregion #LoadData

        #region #SaveDataEventHandler
        private void OnAppointmentChangedInsertedDeleted(object sender, PersistentObjectsEventArgs e) {
            appointmentsTableAdapter.Update(schedulerTestDataSet);
            schedulerTestDataSet.AcceptChanges();
        }
        #endregion #SaveDataEventHandler
        #region #RowUpdatedEventHandler
        private void appointmentsTableAdapter_RowUpdated(object sender, System.Data.OleDb.OleDbRowUpdatedEventArgs e) {
            if (e.Status == UpdateStatus.Continue && e.StatementType == StatementType.Insert) {
                int id = 0;
                using (System.Data.OleDb.OleDbCommand cmd = 
                    new System.Data.OleDb.OleDbCommand("SELECT @@IDENTITY", appointmentsTableAdapter.Connection)) {
                    id = (int)cmd.ExecuteScalar();
                }
                e.Row["ID"] = id;
            }
        }
        #endregion #RowUpdatedEventHandler
    }
}

File: VB/SchedulerDbAccessExample/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports DevExpress.XtraScheduler

Namespace SchedulerDbAccessExample
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
'            #Region "#SaveDataEvent"
            ' ...
            AddHandler schedulerStorage1.AppointmentsChanged, AddressOf OnAppointmentChangedInsertedDeleted
            AddHandler schedulerStorage1.AppointmentsInserted, AddressOf OnAppointmentChangedInsertedDeleted
            AddHandler schedulerStorage1.AppointmentsDeleted, AddressOf OnAppointmentChangedInsertedDeleted
            ' ...
'            #End Region ' #SaveDataEvent
'            #Region "#RowUpdatedEvent"
            ' ...
            AddHandler appointmentsTableAdapter.Adapter.RowUpdated, AddressOf appointmentsTableAdapter_RowUpdated
            ' ...
'            #End Region ' #RowUpdatedEvent
            schedulerControl1.Start = Date.Today
            schedulerControl1.GroupType = SchedulerGroupType.Resource
            schedulerControl1.ActiveView.ResourcesPerPage = 2

            schedulerStorage1.Appointments.ResourceSharing = True

            schedulerControl1.OptionsView.ResourceHeaders.ImageAlign = DevExpress.XtraScheduler.HeaderImageAlign.Top
            schedulerControl1.OptionsView.ResourceHeaders.ImageSizeMode = DevExpress.XtraScheduler.HeaderImageSizeMode.ZoomImage
            schedulerControl1.OptionsView.ResourceHeaders.ImageSize = New Size(140, 70)
            schedulerControl1.OptionsView.ResourceHeaders.Height = 100
        End Sub


        #Region "#LoadData"
        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            ' TODO: This line of code loads data into the 'schedulerTestDataSet.Resources' table. You can move, or remove it, as needed.
            Me.resourcesTableAdapter.Fill(Me.schedulerTestDataSet.Resources)
            ' TODO: This line of code loads data into the 'schedulerTestDataSet.Appointments' table. You can move, or remove it, as needed.
            Me.appointmentsTableAdapter.Fill(Me.schedulerTestDataSet.Appointments)
        End Sub
        #End Region ' #LoadData

        #Region "#SaveDataEventHandler"
        Private Sub OnAppointmentChangedInsertedDeleted(ByVal sender As Object, ByVal e As PersistentObjectsEventArgs)
            appointmentsTableAdapter.Update(schedulerTestDataSet)
            schedulerTestDataSet.AcceptChanges()
        End Sub
        #End Region ' #SaveDataEventHandler
        #Region "#RowUpdatedEventHandler"
        Private Sub appointmentsTableAdapter_RowUpdated(ByVal sender As Object, ByVal e As System.Data.OleDb.OleDbRowUpdatedEventArgs)
            If e.Status = UpdateStatus.Continue AndAlso e.StatementType = StatementType.Insert Then
                Dim id As Integer = 0
                Using cmd As New System.Data.OleDb.OleDbCommand("SELECT @@IDENTITY", appointmentsTableAdapter.Connection)
                    id = DirectCast(cmd.ExecuteScalar(), Integer)
                End Using
                e.Row("ID") = id
            End If
        End Sub
        #End Region ' #RowUpdatedEventHandler
    End Class
End Namespace
